internals-js/src/federation.ts

```diff
@@ -93,10 +94,21 @@ import {
 
 const linkSpec = LINK_VERSIONS.latest();
 const tagSpec = TAG_VERSIONS.latest();
-const federationSpec = FEDERATION_VERSIONS.latest();
+const federationSpec = (version?: FeatureVersion): FederationSpecDefinition => {
+  if (!version) return FEDERATION_VERSIONS.latest();
+  const spec = FEDERATION_VERSIONS.find(version);
+  assert(spec, `Federation spec version ${version} is not known`);
+  return spec;
+};
+
 // Some users rely on auto-expanding fed v1 graphs with fed v2 directives. While technically we should only expand @tag
 // directive from v2 definitions, we will continue expanding other directives (up to v2.4) to ensure backwards compatibility.
-const autoExpandedFederationSpec = FEDERATION_VERSIONS.find(new FeatureVersion(2, 4))!;
+const autoExpandedFederationSpec = federationSpec(new FeatureVersion(2, 4));
```

Comment:
I believe we stopped auto expanding at `2.4` so folks don't get conflicts with `@authenticated` (which seemed like pretty common name for a directive) -> since auto upgrading stuff wouldn't be able to get any of those new directives (they would have to be prefixed with `@federation__` and it is highly unlikely folks using fed v1 would try this), why don't we just stick to 2.4?

Reply:
I'm a little worried that if we've been auto-upgrading people to 2.6+, that there is more risk in going back to `2.4`, but I can't really think of a case where it would be problematic. On the other hand, I don't think there is a risk in going to 2.6 either. If they want the 2.6 specific directives, they would need to explicitly import them.

Reply:
just wondering whether we need to keep track of 3 different fed versions -> latest, auto upgrade to v2.6 and auto provide definitions up to v2.4

since autoupgraded folks are on fed 1 and they cannot use post 2.4 directives without explicit declaration (and they shouldn't be using fed 2 directives in fed 1 anyway) I don't think it matters